graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java

```diff
@@ -213,6 +213,8 @@ private void filterSearch(EventFactory eventFactory, AggregationEventProcessorPa
                 final Message msg = resultMessage.getMessage();
                 final Event event = eventFactory.createEvent(eventDefinition, msg.getTimestamp(), eventDefinition.title());
                 event.setOriginContext(EventOriginContext.elasticsearchMessage(resultMessage.getIndex(), msg.getId()));
+                event.setTimerangeStart(parameters.timerange().getFrom());
+                event.setTimerangeEnd(parameters.timerange().getTo());
```

Comment:
Is there any reason we wouldn't want to have the timerange of the EventProcessorParameters added to the event here? It makes life much easier when trying to generate a search query than trying to create some arbitrary range around the `timestamp` value of the event message. Open to feedback.

Reply:
@kingzacko1 We use the timerange start and end values for events we create via aggregations because a single event represents aggregated values for a timerange. For filtered events, there is no timerange because the event describes a single message.

If we add the parameters timerange here, we change the semantics of the fields. For aggregation events, the timerange is also not always the timerange from the parameters. (see the comment [here](https://github.com/Graylog2/graylog2-server/blob/48b9e8bf5b2ec121bf022a36e3af4586ea516fd8/graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java#L283)) /cc @mpfz0r 

This PR couples the events closer to OpenSearch. (e.g., by adding the query to the event DTOs) How would we handle events that are not created via OpenSearch queries? For example, we are thinking about systems events right now that are made from events in the system and not from messages in OpenSearch.
What do we put into the `query` for Sigma rules queries? Are these always a single query string?

Is there a design document or pitch for this feature? The issue is relatively sparse in detail. If we didn't do that, I think we should take a step back and talk with product management about how the feature should work in general and for events that are not created by OpenSearch searches and what

Reply:
> We use the timerange start and end values for events we create via aggregations because a single event represents aggregated values for a timerange. For filtered events, there is no timerange because the event describes a single message.
>If we add the parameters timerange here, we change the semantics of the fields. For aggregation events, the timerange is also not always the timerange from the parameters. (see the comment [here](https://github.com/Graylog2/graylog2-server/blob/48b9e8bf5b2ec121bf022a36e3af4586ea516fd8/graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java#L283)) /cc @mpfz0r

That makes sense. Prior to the last commit I was actually just searching for the single message in the `origin_context` field, but without time range information the search page would default to 5 minutes and if the message was older than that you wouldn't be able to find it. I was going to fashion a time range around the `timestamp` field, but having a time range that the event processor ran resolved this issue in a way that seemed to make more sense. It also didn't give the users any insight to the actual query that was used to find the message, just the message itself. But for filter events I think that would suffice. Not sure if it is easy to see changes for all commits except the most recent, but if you look at that it doesn't have a query on the EventDTO and relies on either the existing `origin_context` or aggregation `timerange_start/end` fields.

> This PR couples the events closer to OpenSearch. (e.g., by adding the query to the event DTOs) How would we handle events that are not created via OpenSearch queries? For example, we are thinking about systems events right now that are made from events in the system and not from messages in OpenSearch.
What do we put into the query for Sigma rules queries? Are these always a single query string?

I was not considering system events that do not involve OpenSearch queries. Actually didn't even know those were a thing. Would it be possible to create a level of abstraction for events that come from OpenSearch queries and ones that do not? Then one could have the `query` field and the other would not. As Sigma rule queries work now, we generate a single string that is queried. See the dependent PR in the enterprise repo for how that gets set.

>Is there a design document or pitch for this feature? The issue is relatively sparse in detail. If we didn't do that, I think we should take a step back and talk with product management about how the feature should work in general and for events that are not created by OpenSearch searches and what

[Here is the pitch.](https://graylogdocumentation.atlassian.net/wiki/spaces/~6316c47e9794410874c6fb26/pages/2610724902/Sigma+Events+-+Log+search+replay) It was supposed to only be for Sigma Events, but since they use the existing Events framework it seemed to make sense to enable a replay for all events that were generated from an OpenSearch query.
@bernd

Reply:
I think we don't need to add the  `message_replay_info` field to the event index mapping. Or do you think we need to search for any information in the object? It will be available the message JSON we get from OpenSearch.

Reply:
That makes sense to me to avoid changing what the existing timerange fields have been being used for.

Reply:
@bernd I don't think we'll need to search the new `message_replay_info` object for anything. Maybe there is some use case for it, but as far as this feature is concerned we just need it to be returned with the message when searching for the events.

Reply:
Adding a `message_replay_info` sounds like a good idea.

Not sure if you already looked at the existing code, that builds the backlog of messages for notifications.
It kind of is a replay of the search already.
https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java#L140

https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java#L175-L176

But the search relies on the values from the event definition, which can be changed, so having and using the historic info
from the time the event was created, is better. 
We also need to store the configured streams within the `message_replay_info` then.

Reply:
> We also need to store the configured streams within the `message_replay_info` then.

Good idea. I will add that as well!